cmd/run.go

```diff
@@ -83,146 +125,193 @@ Default builder is pack.
 		fmt.Println("internal: error while calling RegisterFlagCompletionFunc: ", err)
 	}
 
+	if err := cmd.RegisterFlagCompletionFunc("builder-image", CompleteBuilderImageList); err != nil {
+		fmt.Println("internal: error while calling RegisterFlagCompletionFunc: ", err)
+	}
+
 	return cmd
 }
 
 func runRun(cmd *cobra.Command, args []string, newClient ClientFactory) (err error) {
-	cfg, err := newRunConfig(cmd)
-	if err != nil {
+	var (
+		cfg runConfig
+		f   fn.Function
+	)
+	if cfg, err = newRunConfig(cmd).Prompt(); err != nil {
 		return
 	}
-
-	function, err := fn.NewFunction(cfg.Path)
-	if err != nil {
+	if err = cfg.Validate(cmd); err != nil {
 		return
 	}
-	if !function.Initialized() {
-		return fmt.Errorf("the given path '%v' does not contain an initialized function", cfg.Path)
+	if f, err = fn.NewFunction(cfg.Path); err != nil {
```

Comment:
I think this is probably where `Initialized()` ends up being checked. I suppose the deleted check earlier in this file was redundant.

Reply:
Right, that is the pattern I'm following in the other commands as well.   So if there is an error in the logic than we "get" to fix a bug in all of them.

Generally speaking, I am trying to keep the CLI as dumb as possible while still being useful and helpful to the user.   The implementation of `Run` is the only part of the equation that actually cares if the function has actually been initialized, so the check is done there.